src/components/src/ImageContainer/image.container.tsx

```diff
@@ -0,0 +1,63 @@
+/**
+ * Copyright © 2019 Elastic Path Software Inc. All rights reserved.
+ *
+ * This is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this license. If not, see
+ *
+ *     https://www.gnu.org/licenses/
+ *
+ *
+ */
+
+import React from 'react';
+import { getConfig, IEpConfig } from '../utils/ConfigProvider';
+
+let Config: IEpConfig | any = {};
+
+
+interface ImageContainerProps {
+  /** name of file */
+  fileName: string;
+  /** image URL */
+  imgUrl: string;
+  /** class name */
+  className?: string;
+}
+
+function ImageContainer(props: ImageContainerProps) {
+  const { fileName, imgUrl, className } = props;
+
+  const epConfig = getConfig();
+  Config = epConfig.config;
+
+  const handleError = (e, defaultImgUrl) => {
+    const { src } = e.target;
+    const types = ['.png', '.svg', '.jpg', '.tif'];
```

Comment:
I might be wrong about this, but I feel like maybe this should live in `ep.config.json`? This would give consumers the ability to do the following without making code changes:

- Control the order in which types are checked.
- Add or remove file types.
- Disable the feature if it winds up hitting render times too hard.

Also, would it be worth filtering `types` to remove whatever initial file type is provided to `props.fileName`?

As this stands now, I'm pretty sure that if `props.fileName` is `foo.png` but the actual file is `foo.jpg`, the component will make a second unnecessary request for `foo.png`, because `.png` appears in the types list before `.jpg`.

Reply:
@plundaahl-ep, if we have an image "banner-1.jpg". and the first request has status code is 400 we don't send more requests with this file type. Then we will take the next file-types in the array. If the image is not on the CDN, then we display the local one.

Reply:
I think I need to update this component

Reply:
Ah! Thanks @petro97, I got thrown off by the regex and I made some incorrect assumptions on my first pass through.

I did some quick modifications of your code to strip it down to just business logic, and I ran through them in my browser console. It looks like the `handleError` function jumps directly to the initially-provided type, meaning that if the `fileName` prop is a `jpg`, it will never try `png` or `svg`.

```
let curSrc;

const handleError = () => {
  const src = curSrc;
  const types = ['.png', '.svg', '.jpg', '.tif'];
  const currentType = src.replace(/.*(?=\.)/g, '');
  const i = types.indexOf(currentType);
  if (types[i + 1]) {
    curSrc = curSrc.replace(currentType, types[i + 1]);
    console.log(curSrc);
  } else {
    console.log('default');
  }
};

curSrc = 'myimg.png';
handleError(); // myimg.svg
handleError(); // myimg.jpg
handleError(); // myimg.tif
handleError(); // default

curSrc = 'myimg.jpg';
handleError(); // myimg.tif
handleError(); // default
```

Reply:
I'm not sure if you want to avoid adding state to this, but if so, maybe something like the following would work?

```
const fallbackTypes = ['.png', '.svg', '.jpg', '.tif'];

function ImageContainer(props: ImageContainerProps) {
  const { fileName, imgUrl, className } = props;

  const initType = fileName.replace(/.*(?=\.)/g, '');
  const types = [
    initType,
    ...fallbackTypes.filter(fileType => fileType !== initType)
  ];

  const handleError = (e, defaultImgUrl) => {
    const { src } = e.target;
    const currentType = src.replace(/.*(?=\.)/g, '');
  // ...
```

Thanks - I'm sure our SCs will appreciate this feature!

Reply:
Thanks @plundaahl-ep, This is actually great idea, I'll use this array filter to check all types of files